addons/netfox/rollback/rollback-synchronizer.gd

```diff
@@ -611,55 +594,64 @@ func _submit_inputs(inputs: Array, tick: int):
 			# Input too old
 			_logger.warning("Received input for %s, rejecting because older than %s frames", [input_tick, NetworkRollback.history_limit])
 			continue
-
-		var input := inputs[offset] as Dictionary
-		if input == null:
+		
+		var input := PropertyStoreSnapshot.deserialize(inputs[offset] as Dictionary)
+		
+		if input.size() == 0:
 			# We've somehow received a null input - shouldn't happen
 			_logger.error("Null input received for %d, full batch is %s", [input_tick, inputs])
 			continue
 
 		var sanitized = _sanitize_by_authority(input, sender)
 
 		if not sanitized.is_empty():
-			var known_input := _inputs.get(input_tick)
+			var known_input := _inputs.get_snapshot(input_tick)
+			# TODO: Does this work with PropertyStoreSnapshot?
```

Comment:
Probably not, as these are compared by reference IIRC. A unit test would decide though :smile: I'm not aware of any way to override the `==` operator in Godot, so our best bet to write a `.equals()` method.